lib/bugsnag/middleware/suggestion_data.rb

```diff
@@ -0,0 +1,27 @@
+module Bugsnag::Middleware
+  class SuggestionData
+
+    CAPTURE_REGEX = /Did you mean\?([\s\S]+)$/
+    DELIMITTER = "\n"
+
+    def initialize(bugsnag)
+      @bugsnag = bugsnag
+    end
+
+    def call(report)
+      matches = {}
+      report.raw_exceptions.each do |exception| 
+        match = CAPTURE_REGEX.match(exception.message)
+        next unless match
+        
+        suggestions = match.captures[0].split(DELIMITTER).each { |suggestion|
+          matches[matches.size] = suggestion.strip
+        }
+      end
+
+      report.add_tab(:"did you mean", matches) if matches.size > 0
```

Comment:
Is there a case for this to happen multiple times in a single exception? Otherwise I think we can better name the metadata field something like `suggestion` so that it is filterable by users rather than using an index.

Reply:
I've changed the tab name to `suggestions`, is there a better method of attaching the suggestions than indexed in a hash?  It seems like the library blocks directly attaching an array as a tab.

Reply:
Naming the tab "did you mean" is fine, the matches could instead be placed in a hash:

```ruby
{suggestions: matches}
```

Though since there's usually one, we should instead unwrap it:

```ruby
if matches.size == 1
  report.add_tab(:"did you mean", {suggestion: matches.first}
elsif matches.size > 1
  # add suggestions array, as above
end
```